src/main/java/com/redhat/devtools/lsp4ij/server/definition/LanguageServerDefinitionListener.java

```diff
@@ -28,6 +29,7 @@ class LanguageServerAddedEvent {
 
         public LanguageServerAddedEvent(@NotNull Collection<LanguageServerDefinition> serverDefinitions) {
             this.serverDefinitions = serverDefinitions;
+            LanguageServiceAccessor.checkCurrentlyOpenFiles();

```

Comment:
the server definitions should be passed to the method, you want only the files affected by the new LS to trigger its startup.